gapic/src/main/com/google/gapid/views/DeviceDialog.java

```diff
@@ -324,6 +358,33 @@ protected void setValidationStatus(DeviceValidationResult result) {
       }
     }
 
+    static String IncompatibilityMessage(Device.Incompatibility incompatibility) {
+      switch (incompatibility.getNumber()) {
+        case Device.Incompatibility.Compatible_VALUE:
+          return "Compatible";
+        case Device.Incompatibility.ABIOS_VALUE:
+          return "Operating System";
+        case Device.Incompatibility.ABIArchitecture_VALUE:
+          return "ABI architecture";
+        case Device.Incompatibility.ABIMemoryLayout_VALUE:
+          return "ABI memory layout";
+        case Device.Incompatibility.GPUVendorID_VALUE:
+          return "GPU Vendor ID";
+        case Device.Incompatibility.GPUDeviceID_VALUE:
+          return "GPU Device ID";
+        case Device.Incompatibility.VkDriverVersion_VALUE:
+          return "Vulkan driver version";
+        case Device.Incompatibility.VkAPIVersion_VALUE:
+          return "Vulkan API version";
+        case Device.Incompatibility.NoTraceVkDriver_VALUE:
+          return "Capture has no Vulkan driver info";
+        case Device.Incompatibility.NoVkSupport_VALUE:
+          return "Device has no Vulkan support";
+        default:
+          return "Unknown incompatibility";
+      }
+    }
+
```

Comment:
I'd prefer we use the `stringtable.Msg` mechanism here to get more details from the server. For example, instead of just saying "ABI architecture", which users will not understand what it means, we can say "Device doesn't support the arm64-v8a ABI", etc. The reality is that this dialog will most often be shown when someone opens a trace that can no longer replay on the one device they have connected (i.e. not to select from multiple devices), so we need to be really clear why we are refusing to open a user's trace.

As it is, we are just translating enums to strings, which is what the stringtable is supposed to do for us.

Reply:
I've laid down the basis to use stringtable, but I'm not sure I'm doing it right -- it seems to slow down the apparition of the pop-up.

Reply:
The incompatibility reason messages are now properly generated on the server side.